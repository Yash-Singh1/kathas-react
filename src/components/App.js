import React from 'react';
import NavigationBar from './NavigationBar';
import KATHAS from '../data/kathas';
import { Link } from 'react-router-dom';

const App = () => (
  <NavigationBar>
    {KATHAS.map((katha, index) => {
      return (
        <div className='card' key={index}>
          <img className='card-image' src={katha.pic} alt={katha.nameEnglish} />
          <p className='card-text'>
            {katha.nameEnglish} / {katha.nameHindi}
          </p>
          <Link
            className='card-button btn btn-primary'
            style={{
              marginLeft: 10
            }}
            to={`/${katha.pathName}/chalisa`}
          >
            Chalisa
          </Link>
          <Link
            className='card-button btn btn-primary'
            style={{
              marginLeft: 10
            }}
            to={`/${katha.pathName}/aarti`}
          >
            Aarti
          </Link>
        </div>
      );
    })}
  </NavigationBar>
);

export default App;
