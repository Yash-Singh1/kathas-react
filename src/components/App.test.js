import React from 'react';
import App from './App';
import KATHAS from '../data/kathas';
import { shallow } from 'enzyme';

describe('App', () => {
  const app = shallow(<App />);

  it('renders properly', () => {
    expect(app).toMatchSnapshot();
  });

  it('contains the navbar', () => {
    expect(app.find('NavigationBar').exists()).toBe(true);
  });

  it('contians the gods', () => {
    expect(app.find('div.card').length).toEqual(KATHAS.length);
  });
});
