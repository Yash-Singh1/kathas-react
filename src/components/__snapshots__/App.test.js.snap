// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly 1`] = `
<NavigationBar>
  <div
    className="card"
    key="0"
  >
    <img
      alt="Durga"
      className="card-image"
      src=""
    />
    <p
      className="card-text"
    >
      Durga
       / 
      दुर्गा
    </p>
    <Link
      className="card-button btn btn-primary"
      style={
        Object {
          "marginLeft": 10,
        }
      }
      to="/durga/chalisa"
    >
      Chalisa
    </Link>
    <Link
      className="card-button btn btn-primary"
      style={
        Object {
          "marginLeft": 10,
        }
      }
      to="/durga/aarti"
    >
      Aarti
    </Link>
  </div>
  <div
    className="card"
    key="1"
  >
    <img
      alt="Ganesh"
      className="card-image"
      src=""
    />
    <p
      className="card-text"
    >
      Ganesh
       / 
      गणेश
    </p>
    <Link
      className="card-button btn btn-primary"
      style={
        Object {
          "marginLeft": 10,
        }
      }
      to="/ganesh/chalisa"
    >
      Chalisa
    </Link>
    <Link
      className="card-button btn btn-primary"
      style={
        Object {
          "marginLeft": 10,
        }
      }
      to="/ganesh/aarti"
    >
      Aarti
    </Link>
  </div>
  <div
    className="card"
    key="2"
  >
    <img
      alt="Hanuman"
      className="card-image"
      src=""
    />
    <p
      className="card-text"
    >
      Hanuman
       / 
      हनुमान
    </p>
    <Link
      className="card-button btn btn-primary"
      style={
        Object {
          "marginLeft": 10,
        }
      }
      to="/hanuman/chalisa"
    >
      Chalisa
    </Link>
    <Link
      className="card-button btn btn-primary"
      style={
        Object {
          "marginLeft": 10,
        }
      }
      to="/hanuman/aarti"
    >
      Aarti
    </Link>
  </div>
  <div
    className="card"
    key="3"
  >
    <img
      alt="Krishna"
      className="card-image"
      src=""
    />
    <p
      className="card-text"
    >
      Krishna
       / 
      कृष्णा
    </p>
    <Link
      className="card-button btn btn-primary"
      style={
        Object {
          "marginLeft": 10,
        }
      }
      to="/krishna/chalisa"
    >
      Chalisa
    </Link>
    <Link
      className="card-button btn btn-primary"
      style={
        Object {
          "marginLeft": 10,
        }
      }
      to="/krishna/aarti"
    >
      Aarti
    </Link>
  </div>
  <div
    className="card"
    key="4"
  >
    <img
      alt="Laxmi"
      className="card-image"
      src=""
    />
    <p
      className="card-text"
    >
      Laxmi
       / 
      लक्ष्मी
    </p>
    <Link
      className="card-button btn btn-primary"
      style={
        Object {
          "marginLeft": 10,
        }
      }
      to="/laxmi/chalisa"
    >
      Chalisa
    </Link>
    <Link
      className="card-button btn btn-primary"
      style={
        Object {
          "marginLeft": 10,
        }
      }
      to="/laxmi/aarti"
    >
      Aarti
    </Link>
  </div>
  <div
    className="card"
    key="5"
  >
    <img
      alt="MahaKali"
      className="card-image"
      src=""
    />
    <p
      className="card-text"
    >
      MahaKali
       / 
      महा काली
    </p>
    <Link
      className="card-button btn btn-primary"
      style={
        Object {
          "marginLeft": 10,
        }
      }
      to="/mahakali/chalisa"
    >
      Chalisa
    </Link>
    <Link
      className="card-button btn btn-primary"
      style={
        Object {
          "marginLeft": 10,
        }
      }
      to="/mahakali/aarti"
    >
      Aarti
    </Link>
  </div>
  <div
    className="card"
    key="6"
  >
    <img
      alt="Ram"
      className="card-image"
      src=""
    />
    <p
      className="card-text"
    >
      Ram
       / 
      राम
    </p>
    <Link
      className="card-button btn btn-primary"
      style={
        Object {
          "marginLeft": 10,
        }
      }
      to="/ram/chalisa"
    >
      Chalisa
    </Link>
    <Link
      className="card-button btn btn-primary"
      style={
        Object {
          "marginLeft": 10,
        }
      }
      to="/ram/aarti"
    >
      Aarti
    </Link>
  </div>
  <div
    className="card"
    key="7"
  >
    <img
      alt="Sai Baba"
      className="card-image"
      src=""
    />
    <p
      className="card-text"
    >
      Sai Baba
       / 
      साई बाबा
    </p>
    <Link
      className="card-button btn btn-primary"
      style={
        Object {
          "marginLeft": 10,
        }
      }
      to="/saibaba/chalisa"
    >
      Chalisa
    </Link>
    <Link
      className="card-button btn btn-primary"
      style={
        Object {
          "marginLeft": 10,
        }
      }
      to="/saibaba/aarti"
    >
      Aarti
    </Link>
  </div>
  <div
    className="card"
    key="8"
  >
    <img
      alt="Saraswati"
      className="card-image"
      src=""
    />
    <p
      className="card-text"
    >
      Saraswati
       / 
      सरस्वती
    </p>
    <Link
      className="card-button btn btn-primary"
      style={
        Object {
          "marginLeft": 10,
        }
      }
      to="/saraswati/chalisa"
    >
      Chalisa
    </Link>
    <Link
      className="card-button btn btn-primary"
      style={
        Object {
          "marginLeft": 10,
        }
      }
      to="/saraswati/aarti"
    >
      Aarti
    </Link>
  </div>
  <div
    className="card"
    key="9"
  >
    <img
      alt="Shani"
      className="card-image"
      src=""
    />
    <p
      className="card-text"
    >
      Shani
       / 
      शनि
    </p>
    <Link
      className="card-button btn btn-primary"
      style={
        Object {
          "marginLeft": 10,
        }
      }
      to="/shani/chalisa"
    >
      Chalisa
    </Link>
    <Link
      className="card-button btn btn-primary"
      style={
        Object {
          "marginLeft": 10,
        }
      }
      to="/shani/aarti"
    >
      Aarti
    </Link>
  </div>
  <div
    className="card"
    key="10"
  >
    <img
      alt="Shiva"
      className="card-image"
      src=""
    />
    <p
      className="card-text"
    >
      Shiva
       / 
      शिव
    </p>
    <Link
      className="card-button btn btn-primary"
      style={
        Object {
          "marginLeft": 10,
        }
      }
      to="/shiva/chalisa"
    >
      Chalisa
    </Link>
    <Link
      className="card-button btn btn-primary"
      style={
        Object {
          "marginLeft": 10,
        }
      }
      to="/shiva/aarti"
    >
      Aarti
    </Link>
  </div>
  <div
    className="card"
    key="11"
  >
    <img
      alt="Vishnu"
      className="card-image"
      src=""
    />
    <p
      className="card-text"
    >
      Vishnu
       / 
      विष्णु
    </p>
    <Link
      className="card-button btn btn-primary"
      style={
        Object {
          "marginLeft": 10,
        }
      }
      to="/vishnu/chalisa"
    >
      Chalisa
    </Link>
    <Link
      className="card-button btn btn-primary"
      style={
        Object {
          "marginLeft": 10,
        }
      }
      to="/vishnu/aarti"
    >
      Aarti
    </Link>
  </div>
</NavigationBar>
`;
